sdk/core/Azure.Core.Experimental/src/Serialization/SerializationExtensions.cs

```diff
@@ -28,7 +29,7 @@ public static class SerializationExtensions
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during deserialization.</param>
         ///<returns>The data converted to the specified type.</returns>
         public static T ToObject<T>(this BinaryData data, ObjectSerializer serializer, CancellationToken cancellationToken = default) =>
-            (T)serializer.Deserialize(data.ToStream(), typeof(T), cancellationToken);
+            (T)serializer.Deserialize(data.ToStream(), typeof(T), cancellationToken)!;
```

Comment:
Same comment applies to these changed lines in this file. Also, given this is a fully-public API, we should probably be checking parameters here; though, I'm on the side of not checking the `this` parameter since an NRE would be thrown, which is what happened if an instance method was called on `this` anyway.

Reply:
Changed the return type to nullable.

cc @JoshLove-msft if he wants to file an issue for argument checking.